src/sentry/replays/tasks.py

```diff
@@ -28,18 +29,26 @@ def delete_recording_segments(project_id: int, replay_id: str, **kwargs: Any) ->
 
 def delete_replay_recording(project_id: int, replay_id: str) -> None:
     """Delete all recording-segments associated with a Replay."""
-    # Delete the segments which are now stored in clickhouse
-    segments_from_metadata = fetch_segments_metadata(project_id, replay_id, offset=0, limit=10000)
-    for segment_metadata in segments_from_metadata:
-        driver = FilestoreBlob() if segment_metadata.file_id else StorageBlob()
-        driver.delete(segment_metadata)
+    # issue delete requests the segment data and model rows - ALL in parallel
+    with ThreadPoolExecutor(max_workers=4) as executor:
```

Comment:
Let's increase the max-workers to 40.  We may be able to go higher but we would need to test to find out.